### PR TITLE
4446 ignore then fail strategy v3

### PR DESCRIPTION
**Pull request by pditommaso**
This PR implements the fail on ignore by using by setting the option `nextflow.enable.failOnIgnore` in the nextflow config file.

### CODE REVIEW

**Comment by bentsherman** [comment]
I would suggest exposing it as `nextflow.failOnIgnore` so as to not confuse it with feature flags. Long term I don't think we should allow feature flags in the config because they are inherent to the pipeline code.

**Comment by bentsherman** [comment]
In other words:
- `nextflow` config scope is used for runtime options and is only relevant in the config file
- feature flags are used for language options and are only relevant in the script

**Comment by pditommaso** [comment]
That was the first option, but then I realised it's not a feature flag but more a workflow (error handling) behaviour, and as such reported in the workflow metadata

File: modules/nextflow/src/main/groovy/nextflow/Session.groovy [hunk 1]
```diff
@@ -866,6 +866,10 @@ class Session implements ISession {
         config.navigate('nextflow.enable.moduleBinaries', false) as boolean
     }
 
+    boolean failOnIgnore() {
+        config.navigate('workflow.failOnIgnore', false) as boolean
```
Comment (bentsherman) [inline modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]:
```suggestion
        config.navigate('nextflow.failOnIgnore', false) as boolean
```

Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]: Check my comment https://github.com/nextflow-io/nextflow/pull/5071#issuecomment-2174019219

Comment (bentsherman) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]: It should go in this config scope: https://nextflow.io/docs/latest/config.html#scope-nextflow

Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]: Disagree

Comment (bentsherman) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]: I'm not talking about feature flags, I'm talking about the `nextflow` config scope which contains the publish retry settings. It makes sense to group them together

Comment (pditommaso) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:870]: yeah, that was somehow temporary. We are going to deprecated `publishDir`, therefore those settings will likely go under `workflow.output.retryPolicy`.

**Pull request merged**